xata/app/routes/index.tsx

```diff
@@ -0,0 +1,121 @@
+import type { FC } from 'react'
+import type { ActionArgs, LoaderArgs } from '@remix-run/node'
+import type { RemixWithXataExampleRecord } from '~/lib/xata.codegen.server'
+import { json } from '@remix-run/node'
+import { Form, useFetcher, useLoaderData } from '@remix-run/react'
+import { getXataClient } from '~/lib/xata.codegen.server'
+
+export const LINKS = [
+  {
+    description: 'Everything you need to know about Xata APIs and tools.',
+    title: 'Xata Docs',
+    url: 'https://docs.xata.io',
+  },
+  {
+    description: 'In case you need to check some Remix specifics.',
+    title: 'Remix Docs',
+    url: 'https://remix.run/docs',
+  },
+  {
+    description:
+      'Maintain your flow by managing your Xata Workspace without ever leaving VS Code.',
+    title: 'Xata VS Code Extension',
+    url: 'https://marketplace.visualstudio.com/items?itemName=xata.xata',
+  },
+  {
+    description: 'Get help. Offer help. Show us what you built!',
+    title: 'Xata Discord',
+    url: 'https://xata.io/discord',
+  },
+]
+
+type TaskComponent = FC<
+  Pick<RemixWithXataExampleRecord, 'id' | 'title' | 'url' | 'description'>
+>
+export const loader = async ({}: LoaderArgs) => {
+  const xata = getXataClient()
+  const links = await xata.db.remix_with_xata_example.getAll()
+
+  return json(links)
+}
+
+export const action = async ({ request }: ActionArgs) => {
+  const xata = getXataClient()
+  const { action, item } = Object.fromEntries(await request.formData())
+
+  if (action === 'delete' && typeof item === 'string') {
+    await xata.db.remix_with_xata_example.delete(item)
+
+    return null
+  }
+
+  if (action === 'create') {
+    await xata.db.remix_with_xata_example.create(LINKS)
+    
+    return null
+  }
+}
+
+const Task: TaskComponent = ({ id, title, url, description }) => {
+  const fetcher = useFetcher()
+
+  return fetcher.submission ? null : (
+    <li key={url}>
+      <a href={url ?? ''} rel="noopener noreferrer" target="_blank">
+        {title}
+      </a>
+      <p>{description}</p>
+      <fetcher.Form method="put">
+        <input type="hidden" name="action" value="delete" />
+        <button type="submit" name="item" value={id}>
```

Comment:
Let's use `intent` instead of a hidden `action` (see #47)

```suggestion
      <fetcher.Form>
        <input type="hidden" name="id" value={id} />
        <button type="submit" name="intent" value="delete">
```